File: HelloWorldOpenMP/hello_v1.cpp
#include <omp.h>
#include <iostream>

void expensive_function()
{
    size_t counter = 0;
    for (size_t j=0; j<10000; j++) {
        for (size_t k=0; k<10000; k++) {
            counter++;
        }
    }
}

int main()
{
    std::cout << "Maximum number of threads is " << omp_get_max_threads() << "\n";

    #pragma omp parallel for
    for (int i=0; i<5000; i++) {
        std::cout << "Calling an expensive function 5000 times, i = " << i << std::endl;
        expensive_function();
    }
}

File: PythonMultiprocessExample/hello_v1.py
from multiprocessing import Pool, cpu_count
import sys


def expensive_function(i):
    print "Calling expensive function 1000 times, i = %d" % i
    sys.stdout.flush()
    counter = 0
    for i in range(0, 1000):
        for j in range(0, 10000):
            counter = counter + 1

if __name__ == '__main__':
    # single threaded version
    #for i in range(0, 1000):
    #    expensive_function(i)

    # parallel version
    pool = Pool(processes=cpu_count())
    pool.map(expensive_function, range(1000))

